nixos/modules/services/security/bitwarden-directory-connector.nix

```diff
@@ -0,0 +1,327 @@
+      home = "${cfg.directory}";
+      group = "${cfg.user}";
+      homeMode = "711";
+    };
+
+    systemd = {
+      timers.bitwarden_directory_connector = {
+        description = "Sync timer for Bitwarden Directory Connector";
+        wantedBy = ["timers.target"];
+        partOf = ["bitwarden_directory_connector.service"];
+        timerConfig = {
+          OnCalendar = cfg.interval;
+          Unit = "bitwarden_directory_connector.service";
+          Persistent = true;
+        };
+      };
+
+      services.bitwarden_directory_connector = {
+        description = "Main process for Bitwarden Directory Connector";
+        wantedBy = ["multi-user.target"];
+        after = ["network-online.target"];
+        wants = [];
+        path = [pkgs.jq];
+
+        environment = {
+          BITWARDENCLI_CONNECTOR_APPDATA_DIR = cfg.directory;
+          BITWARDENCLI_CONNECTOR_PLAINTEXT_SECRETS = "true";
+        };
+
+        serviceConfig = {
+          Type = "oneshot";
+          User = "${cfg.user}";
+          Group = "${cfg.user}";
+          ExecStartPre = pkgs.writeShellScript "bitwarden_directory_connector-config" ''
+            # create the config file
+            ${cfg.package}/bin/${cfg.binary_name} data-file
+            touch -- ${escapeShellArg cfg.directory}/data.json.tmp
+            chmod 600 ${escapeShellArg cfg.directory}/data.json
+            chmod 600 -- ${escapeShellArg cfg.directory}/data.json.tmp
+
+            ${cfg.package}/bin/${cfg.binary_name} config server ${cfg.domain}
+
+            # now login to set credentials
+            export BW_CLIENTID="$(< ${escapeShellArg cfg.secrets.bitwarden.client_path_id})"
+            export BW_CLIENTSECRET="$(< ${escapeShellArg cfg.secrets.bitwarden.client_path_secret})"
+            ${cfg.package}/bin/${cfg.binary_name} login
+
+            jq '.authenticatedAccounts[0] as $account
+              | .[$account].directoryConfigurations.ldap |= $ldap_data
+              | .[$account].directorySettings.organizationId |= $orgID
+              | .[$account].directorySettings.sync |= $sync_data' \
+              --argjson ldap_data ${escapeShellArg ldap_data} \
+              --arg orgID "''${BW_CLIENTID//organization.}" \
+              --argjson sync_data ${escapeShellArg sync_data} \
+              ${escapeShellArg cfg.directory}/data.json \
+              > ${escapeShellArg cfg.directory}/data.json.tmp
+
+            mv -f -- ${escapeShellArg cfg.directory}/data.json.tmp ${escapeShellArg cfg.directory}/data.json
+
+            # final config
+            ${cfg.package}/bin/${cfg.binary_name} config directory 0
+            ${cfg.package}/bin/${cfg.binary_name} config ldap.password --secretfile ${cfg.secrets.ldap}
+          '';
+
+          ExecStart = "${cfg.package}/bin/${cfg.binary_name} sync";
+
+          ExecStartPost = pkgs.writeShellScript "bitwarden_directory_connector-cleanup" ''
+            rm -f -- ${escapeShellArg cfg.directory}/data.json
```

Comment:
You're right about the `--`, assuming the type is changed as you suggested, since `types.path` checks that it starts with `/`, but `escapeShellArg` is still needed, since the path could contain spaces or other shell-meaningful characters. It's rather cursed to do that, yes, but safer to properly quote user-supplied inputs.

Reply:
Done